### PR TITLE
TEST-46 docs: align guides with voice testing best practices

### PR DESCRIPTION
**Pull request by scott-lowe-vapi**
## Description

- Align Evals and Simulations guides with the testing best practices introduced in #1193.
- Clarify tool, audio, and external-state boundaries; replace brittle examples with focused decision checks.
- Replace incomplete CI snippets with a workflow that waits for completion and blocks on critical failures.
- Add risk-based coverage guidance to tutorials, monitoring, and scorecards; correct stale Eval-suite API descriptions through Fern overrides.
- Follow-up for [TEST-46](https://linear.app/vapi/issue/TEST-46/update-docs-for-testing-best-practices).

## Testing Steps

- [ ] Run the app locally using `fern docs dev` or navigate to preview deployment. Local preview stalled during startup; visual review remains pending.
- [ ] Ensure that the changed pages and code snippets work. Static checks passed; no live calls or Eval runs were made.
- [x] `fern check`: 0 errors, 12 existing API-schema warnings.
- [x] MDX/YAML syntax checks for all 13 changed files, changed cURL JSON payloads, new internal links, and `git diff --check`.
- [x] CI example validated with 12 mocked cases covering passing results, failures, cancellation, missing/empty results, timeout, HTTP/JSON errors, and missing configuration.

### CODE REVIEW

**Comment by scott-lowe-vapi** [comment]
* **#1206** <a href="https://app.graphite.com/github/pr/VapiAI/docs/1206?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/VapiAI/docs/1206?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: fern/assistants/structured-outputs-examples.mdx [hunk 1]
```diff
@@ -1215,6 +1215,8 @@ Always test your structured outputs with these scenarios:
 4. **Edge cases** - Boundary values, special characters
 5. **Real conversations** - Actual call recordings or transcripts
 
+Valid JSON doesn't guarantee a correct extraction or judgment. Compare each output with expected values from human-reviewed examples, including known successes and failures. For outcome judgments, record what evidence is available: a reported booking isn't proof of a calendar update. Use synthetic values or remove personal information from reused calls. Follow [judge calibration and result review](/test/run-and-maintain-tests#review-more-than-the-pass-or-fail-label).
```
Comment (stephenvapiai) [inline fern/assistants/structured-outputs-examples.mdx]:
```suggestion
Valid JSON doesn't guarantee a correct extraction or judgment. Compare each output with expected values from human-reviewed examples, including known successes and failures. 

For outcome judgments, record what evidence is available: a reported booking isn't proof of a calendar update. Use synthetic values or remove personal information from reused calls. Follow [judge calibration and result review](/test/run-and-maintain-tests#review-more-than-the-pass-or-fail-label).
```

**Review by stephenvapiai — APPROVED** [review]
I only made one change and that was splitting a large paragraph into two.

File: fern/apis/api/openapi-overrides.yml [hunk 2]
```diff
@@ -998,6 +998,16 @@ components:
         A record of an eval execution, including its target, status, results,
         costs, completion details, and lifecycle timestamps.
       properties:
+        results:
```
Comment (chris-garber-vapi) [inline fern/apis/api/openapi-overrides.yml:1001]: interesting, I don't know if these overrides can/should get backpropagated into comments in the code, haven't looked into how our autogeneration works
